### DISCUSSION TITLE
Page content remains when switching locales

### DISCUSSION THREAD

**Discussion by Egoizt** [requarks/wiki#5204] [Error / Bug Report]
Hello there!

We have the following installation properties:
```
Digital Ocean Droplet 
Wiki.js 2.5.278
Node.js 16.14.2
Docker Container (Linux)
RAM: 3.85 GB
2 CPU
```

Recently we started using multilingual pages and we've faced a trouble switching locales.
When we switch the locale on a multilingual page  the UI texts are successfully replaced with selected language, navigation is also replaced successfully. 
But **page content remains in a default locale**. Locale switch seems to have no effect. 
Page remains in a default language even if we access the page directly with correct locale url (`/en/...`). 
The only way to see the page content in right locale is to edit page in that locale, save and close edit mode. But on next reload it jumps back to default.

I've tried to clear all kind of caches, accessed wiki in browser privacy mode, used Safari and Chrome. Still no luck. There is also no error messages in browser developer console.

I would really appreciate some help.

Thanks in advance!

**Comment by Egoizt** [comment]
I found a workaround. But it doesn't make me happy.

In fact only "old" articles are affected by the problem. Those created when wiki were set up with a single locale.

New articles created after adding multiple locales work fine.

I've found out that of "old" article being deleted and then recreated in same locale local switch begins to work fine. 

Is there any way to force some magic within wiki to make it think of "old" articles as new?

**Reply by Max2983** [reply]
I have also noticed this:
1. Create one page in one locale.
2. Change locale.
3. Create another page in this locale.
4. Switch back to the first locale.
5. Open the first page and insert an internal link ('Insert Link to Page' button in WYSIWYG editor) to the second page and save it.
6. Switch to the second locale.
7. Open the first page, try to open that link. You'll get '404 Page not found'
8. Switch back to the first locale.
9. Try to open that link. You'll get the page opened just fine.

It is confusing. If global locale settings affects the pages when you publish them, it should be noticed in UI. Otherwise it should be regulated in page locale settings (i.e. path section). Another problem solution might be on the core level, because database knows everything to route the link whatever locale is used.
